baked_in.go

```diff
@@ -315,56 +315,56 @@ func isUnique(fl FieldLevel) bool {
 }
 
 // isMAC is the validation function for validating if the field's value is a valid MAC address.
-func isMAC(fl FieldLevel) bool {
+func isMAC(v *Validate, fl FieldLevel) bool {
```

Comment:
Here and below, `v *Validate` param is being added but never used. I'm not understanding why this is the case.

Reply:
Having not taken the time to fully understand the whole validator package myself, it _appears_ that there's something very weird going on where some things happen at the `*Validate` instance level, and others happen at the global level.

I don't think that this PR is the right place to do a massive refactoring of this library to allow everything to happen at the instance level. If we want to do that (and maybe we do!) it should be as a separate PR, where we can consider multiple different ways of making that happen (e.g. making lots of current top-level helper functions instead be methods on that type, which seems cleaner than having those methods take an instance as their first argument).